### ISSUE TITLE
doc build failures and new warnings on main

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#3800]
Some new warnings seem related to the new sklearn release.

Note: doc warnings are flagged as error during doc build

See
https://github.com/nilearn/nilearn/actions/runs/5441381451/jobs/9895271361#step:33:35688

Attached log
[33_Build docs.txt](https://github.com/nilearn/nilearn/files/11945928/33_Build.docs.txt)

The culprits seems to be:

```
2023-07-03T08:27:09.5655542Z /usr/share/miniconda3/envs/testenv/lib/python3.9/site-packages/nilearn/connectome/connectivity_matrices.py:docstring of sklearn.utils._metadata_requests._MetadataRequester.get_metadata_routing:11: WARNING: py:class reference target not found: utils.metadata_routing.MetadataRequest
```

```
2023-07-03T08:27:09.5704758Z /usr/share/miniconda3/envs/testenv/lib/python3.9/site-packages/nilearn/connectome/connectivity_matrices.py:docstring of sklearn.utils._metadata_requests.RequestMethod.__get__.<locals>.func:26: WARNING: py:class reference target not found: pipeline.Pipeline
```

Seems related to this: https://scikit-learn.org/stable/whats_new/v1.3.html#experimental-under-development

EDIT: [suggestion to rely on intersphinx](https://github.com/scikit-learn/scikit-learn/pull/26747#issuecomment-1619128349) does not seem to apply as this is already the case for nilearn.

---

Listing some of the other most frequent warnings below but are not causing a failure:

### 1

```
2023-07-03T06:42:25.5886498Z /usr/share/miniconda3/envs/testenv/lib/python3.9/site-packages/sklearn/svm/_classes.py:32: FutureWarning: The default value of `dual` will change from `True` to `'auto'` in 1.5. Set the value of `dual` explicitly to suppress the warning.
```

Possibly related to 

https://scikit-learn.org/stable/whats_new/v1.3.html#sklearn-svm

### 2

```
2023-07-03T07:27:55.7437431Z /usr/share/miniconda3/envs/testenv/lib/python3.9/site-packages/sklearn/linear_model/_base.py:121: FutureWarning:
2023-07-03T07:27:55.7442173Z 
2023-07-03T07:27:55.7446221Z 'normalize' was deprecated in version 1.2 and will be removed in 1.4. Please leave the normalize parameter to its default value to silence this warning. The default behavior of this estimator is to not do any normalization. If normalization is needed please use sklearn.preprocessing.StandardScaler instead.
```

### 3

EDIT: See https://github.com/nilearn/nilearn/issues/3719

```
2023-07-03T07:44:10.4039818Z /usr/share/miniconda3/envs/testenv/lib/python3.9/site-packages/nilearn/connectome/connectivity_matrices.py:495: FutureWarning:
2023-07-03T07:44:10.4040149Z 
2023-07-03T07:44:10.4041028Z The default strategy for standardize is currently 'zscore' which incorrectly uses population std to calculate sample zscores. The new strategy 'zscore_sample' corrects this behavior by using the sample std. In release 0.13, the default strategy will be replaced by the new strategy and the 'zscore' option will be removed. Please use 'zscore_sample' instead.
```

**Comment by ymzayek** [comment]
Note that number 3 in some cases will keep appearing because it is a default in some functions that don't expose the responsible parameter and I suspect in other cases it is a bug where the parameter is exposed but not passed properly. See #3719
So maybe I would leave that one out of this issue.

**Comment by Remi-Gau** [comment]
ah yeah I remember you reopening this isuse

**Comment by ymzayek** [comment]
I found a temporary workaround so that our full builds can pass. I'll open a PR

**Comment by Remi-Gau** [comment]
Sweet!!

**Comment by Remi-Gau** [comment]
closed by #3801
